client/network/src/transport.rs

```diff
@@ -99,8 +102,22 @@ pub fn build_transport(
 		.upgrade(upgrade::Version::V1Lazy)
 		.authenticate(authentication_config)
 		.multiplex(multiplexing_config)
-		.timeout(Duration::from_secs(20))
-		.boxed();
+		.timeout(Duration::from_secs(20));
+
+	if !memory_only {
```

Comment:
I sent you a DM on Element but out of caution I think the WebRTC should be enabled only if the listening endpoint was provided because some of the network has not upgraded to 0.9.42 which should mean that Identify would fail for these peers as they don't recognize WebRTC. After a while we could remove this extra check and enable it by default.

Reply:
I can read [here](https://github.com/paritytech/polkadot/releases) that Polkadot 0.9.42 is a "high priority" release, and node operators are supposed to upgrade as soon as possible. The release was a month ago.
My opinion is thus that it should be called out as their fault.

Reply:
But there are also downstream users (parachains) which do not follow the same release schedule etc.

Reply:
That's a fair point, any thoughts @bkchr?

I looked at Moonbeam master and it looks like they're still on v0.9.40 which means this change could break their chain unless we upgrade libp2p-identify for v0.9.40. I'm not sure how that process goes though

Reply:
What if they never upgrade? We should enable webrtc for non-validators by default at some point. At least that was the plan in the beginning.

Reply:
I think they will eventually update and we can enable it for non-validators. AFAIR in December we discussed this and decided that waiting for 2-3 release should be enough

Reply:
The plan is to enable it for Westend at least and we can look into enabling it for Kusama/Polkadot as well